posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,322 @@
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+The "-tr" option specifies a properties file which contains package rename data.  The "-td" option specifies a properties file which contains java direct string replacement data.  The "-tf" option specifies a master properties file which links to other properties files.  Each of the linked properties files contains replacement data for text-like files matching specific patterns, as specified in the master properties file.
+
+These three command line options are most important for application and test developers.  Other command line options are available, but are less important.  For example, the option "-tb" is used to specify data for transforming bundle archives, while the option "-ts" is used to specify which target files are to be transformed.  For more information about command line options, run the transformer with either of the options "-usage" or "-help":
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    -usage
+--
+
+=== Transforming server configurations
+
+Transformation of a server configuration is only necessary when "javax" prefixed package names are exposed in the server configuration.  One case is JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations
+
+When running a Jakarta EE 9 enabled application using Open Liberty, new Jakarta EE 9 features must be specified in the server configuration.  In many cases, the new Jakarta EE 9 features use the same short names as the corresponding Jakarta EE 8 features.  In those cases, the Jakarta EE 9 features simply have an updated feature version.  In other cases the new Jakarta EE 9 features update both the feature short name and the feature version.  For example, "cdi-2.0" was updated to "cdi-3.0", while "ejb-3.2" was updated to "enterpriseBeans-4.0".
+
```

Comment:
Does the user need this information if they update the feature names with the test framework? Or is it only needed if they are manually updating or writing their own rules? Should make that clear.